src/generate_conf.rs

```diff
@@ -76,19 +77,39 @@ fn generate_config(
 ) -> Result<(Vec<Interface>, NetworkConfig), anyhow::Error> {
     let network_state = NetworkState::new_from_yaml(&data)?;
 
-    let interfaces = extract_interfaces(&network_state);
-    validate_interfaces(&interfaces, require_mac_addresses)?;
-
     let config = network_state
         .gen_conf()?
         .get("NetworkManager")
         .ok_or_else(|| anyhow!("Invalid NM configuration"))?
         .to_owned();
 
+    let interfaces = extract_interfaces(&network_state, &config);
+    validate_interfaces(&interfaces, require_mac_addresses)?;
+
     Ok((interfaces, config))
 }
 
-fn extract_interfaces(network_state: &NetworkState) -> Vec<Interface> {
+fn extract_interfaces(
+    network_state: &NetworkState,
+    config_files: &NetworkConfig,
+) -> Vec<Interface> {
+    let mut config_files_map: HashMap<String, Vec<String>> = HashMap::new();
+
+    for (_filename, content) in config_files {
```

Comment:
Any reason for the underscore here? 👀

Reply:
Suggestion from the IDE because `_filename` not used. Suggestions are welcome ;)

Reply:
Sorry, I should have given more context. If we prefix a variable with underscore the linter will be happy, however, the variable is still used and it is only "dropped" (i.e. freed) once it is out of scope. Usually, we'd just use only the underscore (`_` instead of `_filename`) in order to discard the value right away.